packages/compass-aggregations/src/modules/aggregation.spec.ts

```diff
@@ -0,0 +1,288 @@
+    expect(hasNextSpy.getCalls().map(x => x.args), 'calls skip with correct args').to.deep.equal([[]]);
+    expect(toArraySpy.getCalls().map(x => x.args), 'calls skip with correct args').to.deep.equal([[]]);
+
+    const {
+      aggregation: { documents, isLast, limit, page }
+    } = store.getState();
+
+    expect(documents).to.deep.equal(mockDocuments);
+    expect(isLast).to.equal(true);
+    expect(page).to.equal(1);
+    expect(limit).to.equal(20);
+  });
+
+  it('cancels an aggregation', async function () {
+    const store = getMockedStore({
+      isLast: false,
+      loading: false,
+      documents: [
+        { id: 5 }, { id: 6 }, { id: 7 }, { id: 8 },
+      ],
+      limit: 4,
+      page: 2,
+    });
+
+    const aggregateMock = new AggregationCursorMock({
+      hasNext: true,
+      documents: [{ id: 9 }, { id: 10 }, { id: 11 }, { id: 12 }],
+      delayInResponse: 1000,
+    });
+
+    store.dispatch({
+      type: DATA_SERVICE_CONNECTED,
+      dataService: new class {
+        aggregate() {
+          return aggregateMock;
+        }
+      }
+    });
+
+    store.dispatch(fetchNextPage() as any);
+    // let it call .aggregate
+    await waitFor(100);
+
+    // now cancel while its fetching data
+    await store.dispatch(cancelAggregation() as any);
+
+    await waitFor(500);
+
+    const { aggregation } = store.getState();
+
+    expect(aggregation.documents, 'correct value for documents').to.deep.equal([]);
+    expect(aggregation.isLast, 'correct value for isLast').to.equal(false);
+    expect(aggregation.page, 'correct value for page').to.equal(2);
+    expect(aggregation.limit, 'correct value for limit').to.equal(4);
+    expect(aggregation.loading, 'correct value for loading').to.equal(false);
+    expect(aggregation.error, 'correct value for error').to.equal('The operation was cancelled.');
+    expect(aggregation.abortController, 'correct value for abortController').to.equal(undefined);
+  });
+
+  describe('paginates data', function () {
+    it('nextPage -> not on last page', async function () {
+      const store = getMockedStore({
+        isLast: false,
+        loading: false,
+        documents: [
+          { id: 5 }, { id: 6 }, { id: 7 }, { id: 8 },
+        ],
+        limit: 4,
+        page: 2,
+      });
+
+      const mockDocuments = [{ id: 9 }, { id: 10 }, { id: 11 }, { id: 12 }];
+      const aggregateMock = new AggregationCursorMock({
+        hasNext: true,
+        documents: mockDocuments,
+      });
+      const skipSpy = spy(aggregateMock, 'skip');
+      const limitSpy = spy(aggregateMock, 'limit');
+      const hasNextSpy = spy(aggregateMock, 'hasNext');
+      const toArraySpy = spy(aggregateMock, 'toArray');
+
+      store.dispatch({
+        type: DATA_SERVICE_CONNECTED,
+        dataService: new class {
+          aggregate() {
+            return aggregateMock;
+          }
+        }
+      });
+
+      await store.dispatch(fetchNextPage() as any);
+
+      await waitFor(500);
```

Comment:
why 500?

Reply:
wait for redux to update the state. little high just to be on the safer side.